contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js

```diff
@@ -0,0 +1,83 @@
+import { contentNodeLastSavedState } from './index';
+import { isTempId } from 'shared/utils';
+import client from 'shared/client';
+
+export function loadSummaryContentNodes(context, params) {
+  return client.get(window.Urls['summarycontentnode-list'](), { params }).then(response => {
+    const contentNodes = response.data;
+    context.commit('ADD_CONTENTNODES', contentNodes);
+    return contentNodes;
+  });
+}
+
+export function loadSummaryContentNode(context, id) {
+  return client
+    .get(window.Urls['summarycontentnode-detail'](id))
+    .then(response => {
+      const contentNode = response.data;
+      context.commit('ADD_CONTENTNODE', contentNode);
+      return contentNode;
+    })
+    .catch(() => {
+      return;
+    });
+}
+
+export function loadContentNodes(context, params) {
+  return client.get(window.Urls['contentnode-list'](), { params }).then(response => {
+    const contentNodes = response.data;
+    context.commit('ADD_CONTENTNODES', contentNodes);
+    return contentNodes;
+  });
+}
+
+export function loadContentNode(context, id) {
+  return client
+    .get(window.Urls['contentnode-detail'](id))
+    .then(response => {
+      const contentNode = response.data;
+      context.commit('ADD_CONTENTNODE', contentNode);
+      return contentNode;
+    })
+    .catch(() => {
+      return;
+    });
+}
+
+/* CONTENTNODE EDITOR ACTIONS */
+export function saveContentNode(context, contentNodeId) {
+  if (context.getters.getContentNodeIsValid(contentNodeId)) {
+    const contentNodeData = contentNodeLastSavedState.getUnsavedChanges(
+      context.getters.getContentNode(contentNodeId)
+    );
+    if (Object.keys(contentNodeData).length) {
+      if (isTempId(contentNodeId)) {
+        if (!contentNodeData.editors || contentNodeData.editors.length === 0) {
+          contentNodeData.editors = [context.rootGetters.currentUserId];
+        }
+        delete contentNodeData.id;
+        return client.post(window.Urls['contentnode-list'](), contentNodeData).then(response => {
+          const contentNode = response.data;
+          context.commit('ADD_CONTENTNODE', contentNode);
+          context.commit('REMOVE_CONTENTNODE', contentNodeId);
+          return contentNode.id;
+        });
+      }
+
+      return client.patch(window.Urls['contentnode-detail'](contentNodeId), contentNodeData).then(response => {
+        // If successful the data will just be true,
+        // so update our last saved state with the current vuex state.
+        if (response.data) {
+          contentNodeLastSavedState.storeLastSavedState(context.getters.getContentNode(contentNodeId));
+        }
+        return null;
+      });
+    }
+  }
+}
+
+export function deleteContentNode(context, contentNodeId) {
```

Comment:
Right now, the way we handle node deletions is setting the parent_id to the channel.trash_tree.id. However, it might make sense to get rid of the trash_tree, add a deleted flag to the model, and just skip over it on loading and on publishing? 

Also, are bulk operations supported as doing this individually has led to deadlocks in the past?

Reply:
Yeah, this was just copied over from the channel actions and I hadn't changed it yet, but I do think this would be a better way to do it - it would also make `restoring` a very simple action (and also reduce mptt move calls that deletion causes).

We can definitely do this in bulk, as with the above - the important thing is making sure we are PATCHing/running update queries on the backend in this case.